state/autocertcache_test.go

```diff
@@ -0,0 +1,44 @@
+package state_test
+
+import (
+	jc "github.com/juju/testing/checkers"
+	"golang.org/x/crypto/acme/autocert"
+	"golang.org/x/net/context"
+	gc "gopkg.in/check.v1"
+
+	statetesting "github.com/juju/juju/state/testing"
+)
+
+type autocertCacheSuite struct {
+	statetesting.StateSuite
+}
+
+var _ = gc.Suite(&autocertCacheSuite{})
+
+func (s *autocertCacheSuite) TestCache(c *gc.C) {
+	ctx := context.Background()
+	cache := s.State.AutocertCache()
+	err := cache.Put(ctx, "a", []byte("aval"))
+	c.Assert(err, jc.ErrorIsNil)
+	err = cache.Put(ctx, "b", []byte("bval"))
+	c.Assert(err, jc.ErrorIsNil)
+
```

Comment:
Perhaps also an explicit test that shows what happens when you specify an existing key? I'm assuming it's OK to overwrite a cache entry?